pepper-apis/src/main/java/org/broadinstitute/ddp/log/SlackAppender.java

```diff
@@ -43,45 +53,69 @@
 
     private boolean canLog = true;
 
-    /**
-     * Used for testing.  Actual deployments use {@link #SlackAppender() the no-arg constructor}
-     */
-    public SlackAppender(Slack slack, String slackHookUrl, String slackChannel) {
-        this.slack = slack;
-        this.slackHookUrl = slackHookUrl;
-        this.channel = slackChannel;
-    }
+    private List<SlackMessagePayload> messagesToSend = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Consider using one of these instead:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ArrayBlockingQueue.html
I would expect less worries with `synchronized`